src/txdb.cpp

```diff
@@ -74,6 +76,15 @@ bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock) {
             LogPrint(BCLog::COINDB, "Writing partial batch of %.2f MiB\n", batch.SizeEstimate() * (1.0 / 1048576.0));
             db.WriteBatch(batch);
             batch.Clear();
+            int crash_simulate = GetArg("-dbcrashratio", 0);
+            if (crash_simulate) {
+                static FastRandomContext rng;
+                if (rng.rand32() % crash_simulate == 0) {
+                    LogPrintf("Simulating a crash. Goodbye.");
+                    sync();
```

Comment:
I've just removed the sync. It seems not needed for testing (I've done reindexes with hundreds of crashes in between).